nixos/modules/services/x11/display-managers/gdm.nix

```diff
@@ -66,6 +66,8 @@ in
 
       enable = lib.mkEnableOption "GDM, the GNOME Display Manager";
 
+      package = lib.mkPackageOption pkgs [ "gnome" "gdm" ] { };
```

Comment:
What is the point of the option? It is a leaf package so it can be trivially replaced with an overlay.

Reply:
I needed to test a patched version recently and once already before in the past. This just makes it easier if you need to and the option is super simple implementation-wise, so it doesn't really add any complexity to the module. 

Also note that overlays for sub-package sets are all but trivial ;)

Reply:
We are going to move `gdm` to top-level [soon](https://github.com/NixOS/nixpkgs/issues/332210#issuecomment-2267503474) :wink: